app/components/GameOverlaySettings.tsx

```diff
@@ -0,0 +1,105 @@
+import { Component } from 'vue-property-decorator';
+import TsxComponent from 'components/tsx-component';
+import { metadata } from 'components/shared/inputs';
+import VFormGroup from 'components/shared/inputs/VFormGroup.vue';
+import { GameOverlayService } from 'services/game-overlay';
+import { Inject } from 'services/core';
+import { $t } from 'services/i18n';
+
+@Component({})
+export default class GameOverlaySettings extends TsxComponent<{}> {
+  @Inject() gameOverlayService: GameOverlayService;
+
+  enabling = false;
+  overlayTransparency = this.gameOverlayService.state.opacity / 100;
+
+  get enableGameOverlay() {
+    return this.gameOverlayService.state.isEnabled;
+  }
+
+  async setEnableGameOverlay(val: boolean) {
+    this.enabling = true;
+    await this.gameOverlayService.setEnabled(val);
+    this.enabling = false;
+  }
+
+  get enablePreview() {
+    return this.gameOverlayService.state.isPreviewEnabled;
+  }
+
+  setEnablePreview(val: boolean) {
+    this.gameOverlayService.setPreviewEnabled(val);
+  }
+
+  get previewMode() {
+    return this.gameOverlayService.state.previewMode;
+  }
+
+  togglePreviewMode() {
+    this.gameOverlayService.setPreviewMode(!this.previewMode);
+  }
+
+  setOverlayTransparency(value: number) {
+    this.overlayTransparency = value;
+    this.gameOverlayService.setOverlayOpacity(value * 100);
+  }
```

Comment:
What's the rationale for having the local state variable `overlayTransparency`.  It seems like you could just have a getter/setter with no local variable.